File: examples/c/loops.c
/* break/continue in real C loops, verified against the zig cc-compiled
 * original on every target. */
#include <stdio.h>

int main(void) {
    int total = 0;
    for (int i = 0; i < 10; i++) {
        if (i % 2 == 0) {
            continue;
        }
        if (i > 7) {
            break;
        }
        total += i;
    }
    printf("total: %d\n", total);

    int n = 0;
    while (n < 40) {
        n += 9;
        if (n % 2 != 0) {
            continue;
        }
        printf("even: %d\n", n);
    }
    return 0;
}

File: examples/py/loops.py
# break/continue in real Python loops — `continue` in a range-for must still
# advance the loop variable on every target.
def main():
    total = 0
    for i in range(10):
        if i % 2 == 0:
            continue
        if i > 7:
            break
        total += i
    print(total)

    n = 0
    while n < 20:
        n += 1
        if n % 5 != 0:
            continue
        print(n)

main()
